src/flows/context/editor.tsx

```diff
@@ -0,0 +1,211 @@
+import React, {
+  FC,
+  createContext,
+  useState,
+  useCallback,
+  useMemo,
+  useContext,
+  useRef,
+  useEffect,
+} from 'react'
+import {EditorType} from 'src/types'
+import {PipeContext} from 'src/flows/context/pipe'
+
+export enum InjectionType {
+  OnOwnLine = 'onOwnLine',
+  SameLine = 'sameLine',
+}
+
+interface InjectionPosition {
+  row: number
+  column: number
+  shouldStartWithNewLine: boolean
+  shouldEndInNewLine: boolean
+}
+
+export interface InjectionOptions {
+  header?: string | null
+  text: string
+  type: InjectionType
+}
+
+export interface EditorContextType {
+  editor: EditorType | null
+  register: (editor: EditorType) => void
+  inject: (options: InjectionOptions) => void
+  calcInjectiontPosition: (type: InjectionType) => Partial<InjectionPosition>
+  updateText: (t: string) => void
+}
+
+const DEFAULT_CONTEXT: EditorContextType = {
+  editor: null,
+  register: _ => {},
+  inject: _ => {},
+  calcInjectiontPosition: _ => ({}),
+  updateText: _ => {},
+}
+
+export const EditorContext = createContext<EditorContextType>(DEFAULT_CONTEXT)
+
+export const EditorProvider: FC = ({children}) => {
+  const [editor, setEditor] = useState<EditorType>(null)
+  const {data, update} = useContext(PipeContext)
+  const {queries, activeQuery} = data
+  const query = queries[activeQuery]
+
+  // Downstream components use these callbacks.
+  // Deeply nested, memoized components.
+  // Avoid vulnerability to memoization ref checks.
+  // (e.g. Sidebar remaining open).
+  const queryText = useRef(() => query.text)
+  useEffect(() => {
+    queryText.current = () => query.text
+  }, [query.text])
+
+  const register = (instance: EditorType) => {
```

Comment:
instead of wrapping and mapping this, just expose it in the provider under a different name

Reply:
will just pass out `setEditor` then